corehq/apps/reports/views.py

```diff
@@ -1104,6 +1120,22 @@ def _render_report_configs(request, configs, domain, owner_id, couch_user, email
     return response.content.decode("utf-8"), excel_attachments
 
 
+def _is_not_accessible(domain, config, couch_user):
+    '''
+    Checks if request is location restricted for the asked report.
+    Returns True if request is restricted for the asked user and the report. Returns False otherwise.
+    '''
+    # Check if report is location safe
+    if report_has_location_filter(config.subreport_slug, domain):
+        return False
+    if not toggles.LOCATION_RESTRICTED_SCHEDULED_REPORTS.enabled(domain):
+        return False
```

Comment:
Can you explain this conditional block?  If I'm reading it correctly, it sounds like this means that without the feature flag enabled, the reports are always accessible.  I'd expect the reports to not be accessible to restricted users _unless_ the feature flag is enabled.

Reply:
This method is used in `conditionally_location_safe` decorator where if a user is location restricted, they will be able to access the reports only if the flag is enabled. If method returns `False` then access will be restricted.

Related tests - https://github.com/dimagi/commcare-hq/pull/32964/files#diff-1bd0dcd0926f8d64a7a2fb1651f5a7fdf8df3f36b6b6f3fb3f6ec1fce6a798ebR66-R75 where user doesn't have `access_all_locations` permission enabled.

Reply:
The usage in the decorator makes sense to me (restricted users can only see the reports if the flag is enabled), but this is called inside `_render_report_configs` and works a little differently.

This function only restricts access to a report if all of the following are true:
* the report has a location filter
* the flag is enabled
* the user has locations assigned
* the user does not have `access_all_locations`

Instead, I think it should only allow access if the user has `access_all_locations` or all of the following are true:
* the flag is enabled
* the report has a location filter
* and the user has locations assigned

Actually it looks like this method guards against calling `config.get_report_content(...)`:
https://github.com/dimagi/commcare-hq/blob/f383e27fc2d991c39f66e92d6ca372f4dd475a82/corehq/apps/saved_reports/models.py#L366-L366
That method already handles a number of error conditions, including access control.  I think it'd make sense to apply this check there instead.

Reply:
https://github.com/dimagi/commcare-hq/pull/32964/commits/865b93cbf553a7112aeac444e5ad4ff91b431313